Makefile

```diff
@@ -75,14 +72,9 @@ DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT),-v /go/src/github.com/docke
 # This allows to set the docker-dev container name
 DOCKER_CONTAINER_NAME := $(if $(CONTAINER_NAME),--name $(CONTAINER_NAME),)
 
-# enable package cache if DOCKER_INCREMENTAL_BINARY and DOCKER_MOUNT (i.e.DOCKER_HOST) are set
-PKGCACHE_MAP := gopath:/go/pkg goroot-linux_amd64:/usr/local/go/pkg/linux_amd64 goroot-linux_amd64_netgo:/usr/local/go/pkg/linux_amd64_netgo
-PKGCACHE_VOLROOT := dockerdev-go-pkg-cache
-PKGCACHE_VOL := $(if $(PKGCACHE_DIR),$(CURDIR)/$(PKGCACHE_DIR)/,$(PKGCACHE_VOLROOT)-)
-DOCKER_MOUNT_PKGCACHE := $(if $(DOCKER_INCREMENTAL_BINARY),$(shell echo $(PKGCACHE_MAP) | sed -E 's@([^ ]*)@-v "$(PKGCACHE_VOL)\1"@g'),)
 DOCKER_MOUNT_CLI := $(if $(DOCKER_CLI_PATH),-v $(shell dirname $(DOCKER_CLI_PATH)):/usr/local/cli,)
 DOCKER_MOUNT_BASH_COMPLETION := $(if $(DOCKER_BASH_COMPLETION_PATH),-v $(shell dirname $(DOCKER_BASH_COMPLETION_PATH)):/usr/local/completion/bash,)
-DOCKER_MOUNT := $(DOCKER_MOUNT) $(DOCKER_MOUNT_PKGCACHE) $(DOCKER_MOUNT_CLI) $(DOCKER_MOUNT_BASH_COMPLETION)
+DOCKER_MOUNT := $(DOCKER_MOUNT) $(DOCKER_MOUNT_CLI) $(DOCKER_MOUNT_BASH_COMPLETION)
```

Comment:
Can we bind-mount `~/.cache` instead?

Reply:
Or ideally buildkit `RUN --mount` cache

Reply:
> Can we bind-mount ~/.cache instead?

I was thinking about that and even made a second commit, like this:
```diff
diff --git a/Makefile b/Makefile
index bbd795bef2..2b983a8075 100644
--- a/Makefile
+++ b/Makefile
@@ -72,9 +72,11 @@ DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT),-v /go/src/github.com/docke
 # This allows to set the docker-dev container name
 DOCKER_CONTAINER_NAME := $(if $(CONTAINER_NAME),--name $(CONTAINER_NAME),)
 
+GOCACHE := $(shell go env GOCACHE)
+DOCKER_MOUNT_GOCACHE := $(if $(GOCACHE),-v $(GOCACHE):/root/.cache/go-build)
 DOCKER_MOUNT_CLI := $(if $(DOCKER_CLI_PATH),-v $(shell dirname $(DOCKER_CLI_PATH)):/usr/local/cli,)
 DOCKER_MOUNT_BASH_COMPLETION := $(if $(DOCKER_BASH_COMPLETION_PATH),-v $(shell dirname $(DOCKER_BASH_COMPLETION_PATH)):/usr/local/completion/bash,)
-DOCKER_MOUNT := $(DOCKER_MOUNT) $(DOCKER_MOUNT_CLI) $(DOCKER_MOUNT_BASH_COMPLETION)
+DOCKER_MOUNT := $(DOCKER_MOUNT) $(DOCKER_MOUNT_GOCACHE) $(DOCKER_MOUNT_CLI) $(DOCKER_MOUNT_BASH_COMPLETION)
 
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
 GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
```

but then I removed it. The problem is this contaminates my `~/.cache/go-build` directory with root-owned files, resulting in potential problems with the `go build` on the host later. I'm not quite sure how to solve this.

An alternative would be to use `make BIND_DIR=. shell` together with setting `GOCACHE=$(pwd)/.cache/go-build`. This way we'll persist the cache in moby root directory, it will still be owned by root but at least not contaminate the user's `~/.cache`.

WDYT @AkihiroSuda?

Reply:
In any case, this can be done as a followup

Reply:
Added `-v docker-dev-cache:/root/.cache`. Results are promising (all hail go-cache!):
```
$ time make dynbinary
...
real	0m50.121s
user	0m1.974s
sys	0m0.402s
$ time make dynbinary
...
real	0m16.167s
user	0m2.088s
sys	0m0.474s
$ docker volume rm docker-dev-cache
docker-dev-cache
$ time make dynbinary
...
real	0m48.478s
user	0m2.057s
sys	0m0.330s
```

Reply:
Commit updated